src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -67,7 +67,10 @@ public void createWorld(World.Environment environment, String name) {
      * @param delay  How long the player should stand still for before teleporting
      */
     public void teleportHome(@NotNull Player player, @NotNull Island island, int delay) {
-        player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().teleportingHome.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+        if (IridiumSkyblock.getInstance().getUserManager().getUser(player).getIsland().get().getId() == island.getId())
```

Comment:
This will cause errors if the player doesnt have an island, instead do something like 
```java
        if (IridiumSkyblock.getInstance().getUserManager().getUser(player).getIsland().map(Island::getId).orElse(0)==island.getId()
```

Reply:
I fixed that while I merged the changes I just pushed to master into this pr tho